### ISSUE TITLE
NumberPrompt returns NaN if a user responses with 0 or "zero"

### ISSUE THREAD

**Issue by benbrown** [Microsoft/botbuilder-js#420]
When a user responds with 0 to a number prompt, prompt.recognized.succeeded is set to true, but prompt.recognized.value is NaN.

**Comment by benbrown** [comment]
This appears to be a problem in the recognizer, the raw value returned by Recognizers.recognizeNumber is:

```
[ { start: 0,
    end: 0,
    resolution: { value: undefined },
    text: '0',
    typeName: 'number' } ]
```

**Comment by cleemullins** [comment]
@JuanAr Hey Juan, any update on this?

**Comment by johnataylor** [comment]
This issue does not reproduce in the C# implementation.

**Comment by johnataylor** [comment]
I'm seeing exactly the same as Ben on the 1.0.1 version of the Node package.

**Comment by johnataylor** [comment]
Incidentally both implementations, Node and C#, behave well and consistently with very large number strings. Its just zero in node that seems broken.

And note its is both '0' and the word 'zero' that produce the undefined value in the resolution.

**Comment by JuanAr** [comment]
This issue was already addressed in Microsoft/Recognizers-Text#508 (merged). The next release of [recognizers-text-number](https://www.npmjs.com/package/@microsoft/recognizers-text-number) will include this fix.

**Comment by JuanAr** [comment]
Latest version of https://www.npmjs.com/package/@microsoft/recognizers-text-number and https://www.npmjs.com/package/@microsoft/recognizers-text-suite includes a fix for this issue.

**Comment by sgellock** [comment]
work around merged and is in 4.0.5 builds.  verified fixed
